src/Text/Gigaparsec.hs

```diff
@@ -140,7 +142,8 @@ Failure .. -- lookAhead does not roll back input consumed on failure
 -}
 lookAhead :: Parsec a -- ^ the parser, @p@, to execute
           -> Parsec a -- ^ a parser that parses @p@ and never consumes input if it succeeds.
-lookAhead = undefined --TODO:
+lookAhead (Parsec p) = Parsec $ \st ok err ->
+  p st (const . (`ok` st)) err
```

Comment:
Lol, we need to find a way to disable HLint's annoying suggestion to _always_ use infix notation

Reply:
Would you prefer a non-point free solution?

Reply:
in this instance yeah. But in general I'm more partial to `flip` than infix. Problem is then HLint stubbornly tells you about it every time

Reply:
Ah well I definitely didn't spend way too long figuring out the point-free solution

Reply:
<pointfree.io>

Reply:
![image](https://media.tenor.com/nYdAr6rffx0AAAAC/an-emoji-disintegrating-in-pain.gif)

Reply:
Fixed in 467bbfb